django/db/models/aggregates.py

```diff
@@ -211,3 +215,47 @@ def __init__(self, expression, sample=False, **extra):
 
     def _get_repr_options(self):
         return {**super()._get_repr_options(), "sample": self.function == "VAR_SAMP"}
+
+
+class JSONArrayAgg(Aggregate):
+    function = "JSON_ARRAYAGG"
+    output_field = JSONField()
+    arity = 1
+
+    def as_sql(self, compiler, connection, **extra_context):
+        if not connection.features.supports_aggregate_filter_clause:
+            raise NotSupportedError(
+                "JSONArrayAgg is not supported on this database backend."
+            )
+        return super().as_sql(compiler, connection, **extra_context)
+
+    def as_sqlite(self, compiler, connection, **extra_context):
+        sql, params = self.as_sql(
+            compiler, connection, function="JSON_GROUP_ARRAY", **extra_context
+        )
+        if (default := self.default) == []:
+            return sql, params
+        count = self.copy()
+        count.__class__ = Count
+        count_sql, count_params = compiler.compile(count)
+        default_sql = ""
+        default_params = () if self.filter is not None else []
+        if default is not None:
+            default_sql, default_params = compiler.compile(default)
+            default_sql = f" ELSE {default_sql}"
+        sql = f"(CASE WHEN {count_sql} > 0 THEN {sql}{default_sql} END)"
+        return sql, count_params + params + default_params
```

Comment:
We might want to add a comment here explain why this is done for future readers as it's not explicit this is required to handle the fact that SQLite's `JSON_GROUP_ARRAY` doesn't return `NULL` when dealing with empty sets.